### PR TITLE
Update to ChainRulesCore 0.10

### PR DESCRIPTION
**Pull request by devmotion**
Supercedes https://github.com/TuringLang/Bijectors.jl/pull/180 and https://github.com/TuringLang/Bijectors.jl/pull/181.

### CODE REVIEW

File: test/ad/distributions.jl [hunk 1]
```diff
@@ -435,17 +433,21 @@
             # Skellam only fails in these tests with ReverseDiff
             # Ref: https://github.com/TuringLang/DistributionsAD.jl/issues/126
             # PoissonBinomial fails with Zygote
+            # Matrix case does not work with Skellam:
+            # https://github.com/TuringLang/DistributionsAD.jl/pull/172#issuecomment-853721493
             filldist_broken = if d.f(d.θ...) isa Skellam
-                (d.broken..., :ReverseDiff)
+                ((d.broken..., :ReverseDiff), (d.broken..., :Zygote, :ReverseDiff))
```
Comment (yebai) [inline test/ad/distributions.jl:439]: Is the redundant `:ReverseDiff` in `(d.broken..., :ReverseDiff)` and `(d.broken..., :Zygote, :ReverseDiff)` intended?

Comment (devmotion) [reply test/ad/distributions.jl:439]: Yes, ReverseDiff still doesn't work for both tested sizes (single sample + two samples) whereas Zygote only fails if there are multiple samples. Upstream changes in SpecialFunctions and Zygote fixed most of the issues in https://github.com/TuringLang/Bijectors.jl/issues/156 (and https://github.com/TuringLang/DistributionsAD.jl/issues/145) but this one is still present.

**Comment by devmotion** [comment]
Test errors are caused by NNlib it seems - I don't understand why we end up with such an old version (0.7.9), IIRC the problem is fixed in recent versions of NNlib.

**Comment by devmotion** [comment]
Ah OK, so 0.7.9 was the latest release without explicit ChainRules dependency: ChainRulesCore 0.10 is pulled in first and since no more recent version of NNlib is compatible with it (well, 0.7.21 is but it requires Julia >= 1.6) the old version with the bug is installed. I'll try to force NNlib 0.7.18 explicitly (latest release which supports ChainRulesCore 0.9 + Julia 1.3).

**Comment by yebai** [comment]
Many thanks, @devmotion.

**Pull request merged**